src/components/ComponentSettingsModal.jsx

```diff
@@ -0,0 +1,467 @@
+          newBaseApiUrl = baseApiUrl
+        }
+      }
+
+      const content = concatFrontMatterMdBody(frontMatter, mdBody);
+      const base64EncodedContent = Base64.encode(content);
+
+      let params = {};
+      if (newFileName !== fileName || prevCategory !== category) {
+        // We'll need to create a new .md file with a new filename
+        params = {
+          content: base64EncodedContent,
+          pageName: newFileName,
+        };
+
+        // If it is an existing file, delete the existing page
+        if (!isNewFile) {
+          await axios.delete(`${newBaseApiUrl}/pages/${fileName}`, {
+            data: {
+              sha,
+            },
+          });
+        }
+        await axios.post(`${newBaseApiUrl}/pages`, params);
+      } else {
+        // Save to existing .md file
+        params = {
+          content: base64EncodedContent,
+          sha,
+        };
+        await axios.post(`${newBaseApiUrl}/pages/${fileName}`, params);
+      }
+      // Refresh page
+      !isNewFile && window.location.reload();
+      
+      let newPageUrl
+      if (type === 'resource') {
+        newPageUrl = `/sites/${siteName}/resources/${category}/${newFileName}`
+      } else if (type === 'page') {
+        newPageUrl = category ? `/sites/${siteName}/collections/${category}/${newFileName}` : `/sites/${siteName}/pages/${newFileName}`
+      }
+      this.setState({ redirectToNewPage: true, newPageUrl })
+    } catch (err) {
+      console.log(err);
+    }
+  }
+
+  changeHandler = (event) => {
+    const { id, value } = event.target;
+    const { pageFilenames, type } = this.props;
+    const pageFilenamesExc = pageFilenames ? pageFilenames.filter((filename) => filename !== this.state.fileName) : null
+    let errorMessage
+    if (type === 'resource') {
+      errorMessage = validateResourceSettings(id, value);
+    } else if (type === 'page') {
+      errorMessage = validatePageSettings(id, value);
+      const newFileName = generatePageFileName(value);
+      if (errorMessage === '' && pageFilenamesExc && pageFilenamesExc.includes(newFileName)) {
+        errorMessage = 'This title is already in use. Please choose a different one.';
+      }
+    }
+
+    this.setState({
+      errors: {
+        [id]: errorMessage,
+      },
+      [id]: value,
+    });
+  }
+
+  render() {
+    const {
+      category,
+      title,
+      date,
+      allCategories,
+      permalink,
+      fileUrl,
+      errors,
+      canShowDeleteWarningModal,
+      sha,
+      redirectToNewPage,
+      newPageUrl,
+      thirdNavTitle,
+      isPost,
+    } = this.state;
+    const { settingsToggle, isNewFile, type, modalTitle } = this.props;
+
+    // Settings form has errors - disable save button
+    const hasErrors = _.some(errors, (field) => field.length > 0);
+
+    return (
+      <>
+        <div className={elementStyles.overlay}>
+          { (sha || isNewFile)
+          && (
+          <div className={elementStyles.modal}>
```

Comment:
I feel like the current modal is a little too wide, actually removing the height and width restrictions makes it look a bit more aesthetic (and also remove the need to scroll vertically) and also more like how it looks like in our Figma designs:
_after removing height and width constraints_
<img width="646" alt="Screenshot 2020-11-04 at 4 55 43 AM" src="https://user-images.githubusercontent.com/31984694/98039769-30d3aa00-1e5a-11eb-84b3-5476874dc4cf.png">

_figma design_
<img width="575" alt="Screenshot 2020-11-04 at 4 57 40 AM" src="https://user-images.githubusercontent.com/31984694/98039858-5791e080-1e5a-11eb-8c3b-a266b10f2959.png">

But also i recognize that this class is being used elsewhere - so maybe we can create a new class to implement the change?

Reply:
Resolved in https://github.com/isomerpages/isomercms-frontend/pull/184/commits/681061f4bcce4d15edcaff922b2a594b6483324c and https://github.com/isomerpages/isomercms-frontend/pull/184/commits/2ade92096720ba9225eee511bb0d38200db3656c as removing width entirely caused the modal to be too small for the page settings modal. Am open to further suggestions on the design.